src/configure/templateInputHelper/utilities/DataSourceUtility.ts

```diff
@@ -39,7 +40,7 @@ export class DataSourceUtility {
         var httpMethod = dataSource.httpMethod || "GET";
         var requestBody = !!dataSource.requestBody ? MustacheHelper.render(dataSource.requestBody, view) : null;
         let amrClient = new ArmRestClient(azureSession);
-        return amrClient.fetchArmData(armUri, httpMethod, requestBody)
+        return amrClient.fetchArmData(armUri, httpMethod, JSON.parse(requestBody))
```

Comment:
why parse?

Reply:
MustacheHelper.render returns string, and the RestClient that we use for making API calls does stringify of body. If JSON object is not passed then it will become invalid string enclosed in two double quotes after stringify.